packages/frontend/src/features/sqlRunner/components/visualizations/barChartBizLogic.ts

```diff
@@ -0,0 +1,200 @@
+                        'quarter',
+                        'year',
+                    ],
+                });
+                break;
+            case DimensionType.TIMESTAMP:
+                options.push({
+                    type: 'time',
+                    columnId: column.id,
+                    timeGrainOptions: [
+                        'minute',
+                        'hour',
+                        'day',
+                        'week',
+                        'month',
+                        'quarter',
+                        'year',
+                    ],
+                });
+                break;
+            case DimensionType.STRING:
+            case DimensionType.NUMBER:
+            case DimensionType.BOOLEAN:
+                options.push({
+                    type: 'categorical',
+                    columnId: column.id,
+                    timeGrainOptions: [],
+                });
+                break;
+        }
+    }
+    return options;
+};
+
+// available yValues
+type yLayoutOptions = {
+    columnId: string;
+    aggregationOptions: string[];
+};
+const yLayoutOptions = (columns: SqlColumn[]): yLayoutOptions[] => {
+    let options: yLayoutOptions[] = [];
+    for (const column of columns) {
+        switch (column.type) {
+            case DimensionType.NUMBER:
+                options.push({
+                    columnId: column.id,
+                    aggregationOptions: [
+                        'none',
+                        'min',
+                        'max',
+                        'sum',
+                        'average',
+                        'count',
+                        'count_distinct',
+                    ],
+                });
+                break;
+            case DimensionType.STRING:
+            case DimensionType.BOOLEAN:
+                options.push({
+                    columnId: column.id,
+                    aggregationOptions: ['count', 'count_distinct'],
+                });
+                break;
+        }
+    }
+    return options;
+};
+
+// available series
+type seriesLayoutOptions = {
+    columnId: string;
+};
+const seriesLayoutOptions = (columns: SqlColumn[]): seriesLayoutOptions[] => {
+    let options: seriesLayoutOptions[] = [];
+    for (const column of columns) {
+        switch (column.type) {
+            case DimensionType.STRING:
+            case DimensionType.BOOLEAN:
+                options.push({
+                    columnId: column.id,
+                });
+                break;
+        }
+    }
+    return options;
+};
+
+// data transform
+type BarChartSqlTransform = {
+    x: {
+        columnId: string;
+        timeGrain: string;
+        type: 'time' | 'categorical';
+    };
+    y: {
+        columnId: string;
+        aggregation: string;
+        sort: string | null;
+    }[];
+    colorBy: {
+        columnId: string;
+    };
+};
+
+const transform = (config: BarChartConfig, rows: ResultRow[]) => {};
```

Comment:
can you clarify what `transform` would do? Would it use a current transformer, like BarChartDataTransformer?